### DISCUSSION TITLE
What is the chunk size when using streams?

### DISCUSSION THREAD

**Discussion by Vic152** [rabbitmq/rabbitmq-stream-dotnet-client#224] [Q&A]
Hi all,

I have problem establishing how many messages fit in one chunk when using streams. Is it memory based? Does it depend on the message size?

**Comment by kjnilsson** [comment]
It depends on ingress rates. The faster you publish into a stream the
bigger the chunks.

Cheers
Karl

On Wed, 1 Feb 2023 at 16:21, Vic ***@***.***> wrote:

> Hi all,
>
> I have problem establishing how many messages fit in one chunk when using
> streams. Is it memory based? Does it depend on the message size?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/rabbitmq/rabbitmq-stream-dotnet-client/discussions/224>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAJAHFALWO2CU3IXENN6SE3WVKEQPANCNFSM6AAAAAAUN5YIDE>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***
> com>
>


-- 
*Karl Nilsson*

**Reply by Vic152** [reply]
Oh interesting. So it's not based on memory or number of messages. Can we say it's dynamic based on the ingress? Is there a min or max size?

**Reply by Vic152** [reply]
Also this so it is clear once for all: https://github.com/rabbitmq/rabbitmq-website/issues/1598

**Comment by Gsantomaggio** [comment]
closing!
